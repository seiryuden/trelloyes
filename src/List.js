import React from "react";
import Card from "./Card"
import "./List.css"


function List(props){
    
    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>
            <button onClick= {() => props.onAddRandom(props.listKey)}>Add random card</button>
            <div className="List-cards">
                {props.cards.map(function(card, i){
                    
                  return <Card 
                    key={i}
                    listKey= {props.listKey}
                    cardId= {props.cardIds[i]} 
                    title={card.title} 
                    content={card.content}
                    onDeleteItem={props.onDeleteItem}
                    onAddRandom={props.onAddRandom}
                    
                    />
                    
                }
                )}
            </div>
        </section>
    );

}

export default List;