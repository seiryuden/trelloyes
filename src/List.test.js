import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import List from "./List";

describe("List component", () =>{ 

it ("renders without crashing", () =>{

    const cards = [{title: 'First card', content: 'lorem ipsum'}, { title: 'Second card', content: 'lorem ipsum' }];
    const div = document.createElement("div");
    ReactDOM.render(<List cards={cards}/>, div);
    ReactDOM.unmountComponentAtNode(div);

});


it ("renders UI as expected", () =>{
    const cards = [{title: 'First card', content: 'lorem ipsum'}, { title: 'Second card', content: 'lorem ipsum' }];
    const tree = renderer
        .create(<List cards={cards} header="Second List" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    
});
});