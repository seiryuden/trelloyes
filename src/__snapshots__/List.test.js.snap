// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;

exports[`renders UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
